spec/models/legal_aid_application_spec.rb

```diff
@@ -89,4 +89,34 @@
       expect(application.benefit_check_result.dwp_ref).to eq(benefit_check_response[:confirmation_ref])
     end
   end
+
+  describe '#generate_secure_id' do
+    let(:legal_aid_application) { create :legal_aid_application }
+    let(:secure_data) { SecureData.last }
+
+    subject { legal_aid_application.generate_secure_id }
+
+    it 'generates a new secure data object' do
+      expect { subject }.to change { SecureData.count }.by(1)
+    end
+
+    it 'returns the generated id' do
+      expect(subject).to eq(secure_data.id)
+    end
+
+    it 'generates data that can be used to find application' do
+      id = subject
+      secure_data = SecureData.find(id)
+      expect(described_class.find_by(secure_data.retrieve[:find_by])).to eq(legal_aid_application)
```

Comment:
This should be failing, shouldn't it?

Reply:
Weirdly `LegalAidApplication.find_by nil` is returning an instance, and as there is only one instance, it's the matching instance.